specification/monitor/resource-manager/microsoft.insights/preview/2017-12-01-preview/metricNamespaces_API.json

```diff
@@ -0,0 +1,166 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "MonitorManagementClient",
+    "x-ms-code-generation-settings": {
+      "name": "MonitorManagementClient"
+    },
+    "version": "2017-12-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{resourceUri}/providers/microsoft.insights/metricNamespaces": {
+      "get": {
+        "tags": [
+          "metricNamespaces"
+        ],
+        "operationId": "MetricNamespaces_List",
+        "description": "Lists the metric namespaces for the resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/StartTimeParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get the list of metric namespaces",
+            "schema": {
+              "$ref": "#/definitions/MetricNamespaceCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Get Metric Namespaces without filter": { "$ref": "./examples/GetMetricNamespaces.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "MetricNamespaceName": {
+      "properties": {
+        "metricNamespaceName": {
+          "type": "string",
+          "description": "the metric namespace name."
+        }
+      },
+      "description": "The fully qualified metric namespace name."
+    },
+    "MetricNamespace": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "the id of the metricNamespace."
+        },
+        "type": {
+          "type": "string",
+          "description": "the type of the namespace."
+        },
+        "name": {
+          "type": "string",
+          "description": "the name of the namespace."
+        },
+		"properties": {
+          "$ref": "#/definitions/MetricNamespaceName",
```

Comment:
Is there a reason you chose not to use x-ms-client-flatten here?
Using flatten can make for a cleaner client experience, especially as there's only one property in MetricNamespaceName.
However if you expect any of the properties to have naming conflicts (e.g. "name", "type", "id") then it should be left as is.

Whether you've flattened other properties throughout your SDK can also influence your choice here as it would be best to be consistent.

Reply:
I will keep the way it is, we might need add more property under this in the future.